### PR TITLE
Add Server-Timing API as middleware

### PR DESCRIPTION
**Pull request by PassiDel**
This adds [Server-Timing](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Server-Timing) as a middleware.

Please see honojs/middleware#99 for context.

I've replaced `process.hrtime()` with this, since `jest` failed to access performance:
```ts
const getTime = () => {
  try {
    return performance.now()
  } catch {}
  return Date.now()
}
```

This technique may not fully work for cloudflare worker, because they [disallow the use of timers](https://developers.cloudflare.com/workers/learning/security-model/#step-1-disallow-timers-and-multi-threading).

### CODE REVIEW

File: deno_dist/middleware/timing/index.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    }
+  }
+}
+
+interface Timer {
+  description?: string
+  start: number
+}
+
+interface Config {
```
Comment (yusukebe) [inline deno_dist/middleware/timing/index.ts:19]: It's better to be `TimingOptions` like a`CORSOptions`.

File: deno_dist/middleware/timing/index.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+const getTime = () => {
+  try {
+    return performance.now()
+  } catch {}
+  return Date.now()
+}
+
+export const timing = (config?: Partial<Config>): MiddlewareHandler => {
```
Comment (yusukebe) [inline deno_dist/middleware/timing/index.ts:34]: It also should be `options?: Partial<TimingOptions>`

**Review by yusukebe — COMMENTED** [review]
Thank you, @PassiDel!

This is great, even though it can't measure in nanoseconds on Cloudflare Workers, it's still useful.

Screenshot 2023-07-02 at 22 20 23

I've left some comments, please check them out.

**Comment by PassiDel** [comment]
Nice, I wasn't sure how CF Workers would handle that, thanks for checking!

**Comment by yusukebe** [comment]
Thanks! I'll merge it and release `v3.3.0` includes this feature soon. Great work!

**Pull request merged**